### PR TITLE
chore: remove data layer feature

### PR DESCRIPTION
**Pull request by rb-determined-ai**
Remove the data layer feature, which was deprecated in 0.18.0 (May 2022).

## Commentary

This is the sort of change which frequently breaks the rest api and/or webui because we're removing something from the experiment config.

I have taken the following steps:
- [x] Unit tests to make sure old experiment configs still pass our schema
- [x] Unit tests to make sure the master parsing code also allows old configs
- [x] I _think_ I set up the webui to have some sort of equivalent test, by leaving the data_layer in the the OldExperiment.config field.
- [x] manually verified that old data layer experiments can be viewed in the webui and forked succesfully (config-wise... the code wouldn't actually run since the feature is gone)

### CODE REVIEW

File: harness/determined/common/schemas/expconf/_gen.py [hunk 1]
```diff
@@ -1051,14 +783,12 @@
             "default": {}
         },
         "data_layer": {
+            "$comment": "the data_layer feature is gone, and the config is ignored",
```
Comment (ioga) [inline harness/determined/common/schemas/expconf/_gen.py:786]:
```suggestion
            "$comment": "the data_layer feature was removed in 0.19.10, and the config is ignored"
```

**Review by ioga — APPROVED** [review]

**Review by keita-determined — COMMENTED** [review]
i did ad hoc test and it worked well. 
i found some files which use `dataLayer` or `data_layer`. idk if its related to this tho

here are some examples 

https://github.com/determined-ai/determined/blob/7711ef9dcb55d51479a92d78b61e6ae85022f1f5/webui/react/src/utils/task.ts#L84
https://github.com/determined-ai/determined/blob/464f68c4bd991de3fbf81b9cb741c908f5df7eb9/webui/react/src/pages/ExperimentDetails/ExperimentDetails.test.mock.ts#L14
https://github.com/determined-ai/determined/blob/464f68c4bd991de3fbf81b9cb741c908f5df7eb9/webui/react/src/pages/ExperimentDetails/ExperimentDetails.test.mock.ts#L684

**Comment by rb-determined-ai** [comment]
@keita-determined I intentionally left tests with data layer configured.  My assumption is that since experiment configs in the wild can still contain data layer (mostly old ones out of old databases) that the webui should still test against experiment configs with that field set.

Is my reasoning faulty?

**Comment by keita-determined** [comment]
> @keita-determined I intentionally left tests with data layer configured. My assumption is that since experiment configs in the wild can still contain data layer (mostly old ones out of old databases) that the webui should still test against experiment configs with that field set.
> 
> Is my reasoning faulty?

no, make sense. LGTM

**Pull request merged**